app/lib/service/secret/backend.dart

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:gcloud/db.dart';
+import 'package:gcloud/service_scope.dart' as ss;
+
+import '../../shared/exceptions.dart';
+
+import 'models.dart';
+
+export 'models.dart' show SecretKey;
+
+/// Sets the secret backend service.
+void registerSecretBackend(SecretBackend backend) =>
+    ss.register(#_secretBackend, backend);
+
+/// The active secret backend service.
+SecretBackend get secretBackend => ss.lookup(#_secretBackend) as SecretBackend;
+
+/// Represents the backend for the secret handling and related utilities.
+class SecretBackend {
+  final DatastoreDB _db;
+
+  SecretBackend(this._db);
+
+  /// Loads the Secret value from the Datastore.
+  /// Returns null if no Secret entity is found.
+  /// Throws InvalidInputException if the [id] is no valid.
+  Future<String> lookup(String id) async {
+    InvalidInputException.check(SecretKey.isValid(id), '$id is not valid');
```

Comment:
No, this is not an invalid input exception, this kind of flaw should never cause our app to return 4xx.

Do
```dart
if (!SecretKey.isValid(id)) {
  throw ArgumentError.value(id, 'id', 'invalid secret key identifier');
}
``` 

This should never happen, we should never catch it, therefore it should be an error and not an exception.
If it propagates to our request handler (in some way) it should be a 5xx.